src/rpcdataacq.cpp

```diff
@@ -444,183 +469,185 @@ UniValue rpc_exportstats(const UniValue& params, bool fHelp)
+
+            Output << (sum_diff / samples) << " " << (samples / invsum_diff)
+                << " " << min_diff << " " << max_diff << "  ";
             Output << (sum_spacing / samples) << " " << min_spacing << " " << max_spacing << "  ";
-            Output << (sum_size / samples) << " " << min_size << " " << max_size << "  ";
-            Output << (sum_research / samples) << " " << (sum_research / samples_w_research) << " " << max_research << "  ";
-            Output << (sum_interest / samples) << " " << max_interest << "  ";
-            Output << (cnt_empty / samples) << " " << cnt_empty << "  ";
-            Output << (cnt_investor / samples) << " " << cnt_investor << "  ";
-            Output << (cnt_trans / samples) << " " << (cnt_trans / samples_w_trans) << " " << cnt_trans << "  ";
-            Output << (cnt_research / samples) << " " << cnt_research << "  ";
-            Output << (cnt_contract / samples) << " " << cnt_contract << "  ";
-            Output << (cnt_neuralvote / samples) << " " << cnt_neuralvote << "  ";
-            Output << (cnt_neuralcurr / samples) << " " << cnt_neuralcurr << "  ";
-            Output << (sum_magnitude / samples_w_cpid) << "  ";
-            // missing: trans, empty, size, neural
+            Output << sum_size << " " << max_size << "  ";
+
+            Output << sum_research << " " << cnt_research << " " << max_research << "  ";
+            Output << sum_magnitude << "  ";
+            Output << sum_interest << " " << max_interest << "  ";
+
+            Output << cnt_empty << " " << cnt_investor << " " << cnt_bcontract
+                << " " << cnt_neuralvote << " " << cnt_neuralcurr << "  ";
 
+            Output << cnt_trans << "  ";
+            // missing: contract_trans, netweight, stake_output, trx_output
             Output << "\n";
-            samples = 0;
-            sum_diff = 0;
-            min_diff = INT_MAX;
-            max_diff = 0;
-            cnt_empty = 0;
-            cnt_investor = 0;
-            sum_spacing = 0;
-            min_spacing = INT_MAX;
-            max_spacing = 0;
-            sum_size = 0;
-            min_size = INT_MAX;
-            max_size = 0;
-            cnt_trans = 0;
-            sum_research = 0;
-            max_research = 0;
-            sum_interest = 0;
-            max_interest = 0;
-            sum_magnitude = 0;
-            cnt_research = 0;
-            cnt_neuralvote = 0;
-            cnt_neuralcurr = 0;
-            cnt_contract = 0;
         }
-        /* This is wery important */
-        cur = cur->pprev;
+
     }
 
     result1.pushKV("file", o_path.string());
     result1.pushKV("points",(uint64_t)points);
     result1.pushKV("smoothing",smoothing);
     result1.pushKV("blockcount",blockcount);
     Output.close();
+
+    /* Relax permissions of the file as it is not confidential. We set umask
+     * 077 in AppInit2 so by default the permissions are too restrictive. */
+    chmod(o_path.string().c_str(), 0644);
```

Comment:
Looks like chmod isn't available on Windows.

Reply:
You can use boost to change the permissions: https://stackoverflow.com/questions/16810982/how-to-change-file-permissions-using-the-boost-library